File: backend/src/main/java/cn/aksu/supervision/message/entity/Message.java
package cn.aksu.supervision.message.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "message")
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(name = "msg_type", nullable = false, length = 20)
    private String type;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column
    @Builder.Default
    private Boolean isRead = false;

    @Column(length = 500)
    private String link;

    @Column
    private Long relatedId;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createTime;
}

File: android-app/app/src/main/java/cn/aksu/supervision/data/model/Message.kt
package cn.aksu.supervision.data.model

import java.io.Serializable

data class Message(
    val id: Long,
    val title: String,
    val content: String,
    val type: String,           // SYSTEM / TASK / ALERT / CHAT
    val isRead: Boolean,
    val senderId: Long?,
    val senderName: String?,
    val relatedId: Long?,       // 关联业务ID
    val relatedType: String?,   // 关联业务类型
    val createdAt: String
) : Serializable {

    fun typeLabel(): String = when (type) {
        "SYSTEM" -> "系统通知"
        "TASK" -> "任务通知"
        "ALERT" -> "预警通知"
        "CHAT" -> "工作消息"
        else -> type
    }
}
